lib/evmone/instructions.hpp

```diff
@@ -429,28 +548,41 @@ inline void calldatasize(StackTop stack, ExecutionState& state) noexcept
 
 inline Result calldatacopy(StackTop stack, int64_t gas_left, ExecutionState& state) noexcept
 {
-    const auto& mem_index = stack.pop();
-    const auto& input_index = stack.pop();
-    const auto& size = stack.pop();
+    const auto& mem_index = stack[0];
+    const auto& input_index = stack[1];
+    const auto& size = stack[2];
 
-    if (!check_memory(gas_left, state.memory, mem_index, size))
+    if (!check_memory(gas_left, state.memory, mem_index.val, size.val))
```

Comment:
I think we need to add the requirements before calling `check_memory`, otherwise we will not have the requirements when it returns false.

Reply:
hmm good question. I was wondering that... my reasoning was that if check_memory fails, we  abort with EVMC_OUT_OF_GAS so it doesn't matter what the requirements were... but thinking about it, I don't know how the different calls work and it may be that we still want to collect the requirements because the abort only drops us back to the caller frame where we continue?

Reply:
We need the requirements to verify that we will also abort with `EVMC_OUT_OF_GAS` with a different concrete storage configuration.